common/common.go

```diff
@@ -133,26 +137,24 @@ func BlockIngestor(cache *BlockCache, startHeight int, rep int) {
 				}
 			}
 			// We're up to date in our polling; wait for a new block
+			c.Sync()
+			waiting = true
 			Sleep(10 * time.Second)
 			continue
 		}
 		retryCount = 0
 
-		if (height % 100) == 0 {
+		if waiting || (height%100) == 0 {
 			Log.Info("Ingestor adding block to cache: ", height)
 		}
-		reorg, err := cache.Add(height, block)
-		if err != nil {
-			Log.Fatal("Cache add failed")
-		}
 
 		// Check for reorgs once we have inital block hash from startup
-		if reorg {
+		if c.LatestHash != nil && !bytes.Equal(block.PrevHash, c.LatestHash) {
 			// This must back up at least 1, but it's arbitrary, any value
 			// will work; this is probably a good balance.
-			height -= 2
-			reorgCount++
-			if reorgCount > 10 {
+			height = c.Reorg(height - 2)
+			reorgCount += 2
```

Comment:
It's 2 because I changed the definition of `reorgCount` to blocks instead of attempts. That makes the code immediately below make more sense:
```
if reorgCount > 100 {
	Log.Fatal("Reorg exceeded max of 100 blocks! Help!")
}
```